### ISSUE TITLE
Extending the class to check rights on jobs

### ISSUE THREAD

**Issue by smassonnat** [cristal-ise/kernel#52]
Would it be possible to allow the class that calculates/checks Jobs for Roles/Agent (jobPusher() ?) to be extended from a module.
This way it will be possible to redefine the way rights are checked for executing activities' jobs.

**Comment by ogattaz** [comment]
@Andrew: You explained us yesterday that we could implement our AccessRights calculation in the method "hasRole" of  the lookup manager :
```java
	/**
	 * Returns all of the Agents in this centre who hold this role (including sub-roles)
	 * 
	 * @param agentPath
	 * @param role
	 * @return
	 */
	public boolean hasRole(AgentPath agentPath, RolePath role);
```

To be able to implement our algorithm, we need to have an access to the context of te activity for which we check the roles of the agent.
To have this context I wonder if we have the a reference to the activity in the arguments of tha method.

Question: could we retrieve the root parent item of an activity to retrieve its properties (The properties of the process instance in our case) ?

**Comment by abranson** [comment]
I'm not sure how much can be passed from all the different places that hasRole is called. I'll look into what might be possible.

**Comment by ogattaz** [comment]
The goal of this request is to manage delegations : during a period a delegation contract is passed between a delegator and a delegated user. the delegation contract specifies that its scope is all the jobs or a subset of them
During this period, the delegated user has to "see" the job list of his delegator or a subset of that list.
When the delegated user executes one job of his delegator the access right manager have to accept the request

We can see two sub issues : 
1) how to get the job list of an other agent (the delegator) filtered on roles and/or domain path ot the "model" of the item parent of the job ?
2) how to tell Cistal that the execution request of a job of a delegator is acceptable ? by passing an authenticator of the delegator in the request  ?

**Comment by abranson** [comment]
@ogattaz Delegator agent support added to Event in 5bab9777d2209d42e1cf21b611db36080abdc793. There's no way to set it yet though.

**Comment by abranson** [comment]
Implemented in commit:89bc920166b41527a66ba03c8474530695329091
